### ISSUE TITLE
mobile controllstick

### ISSUE THREAD

**Issue by jack-oid** [ScratchAddons/ScratchAddons#7012]
built in mobile control stick. the option to turn it on/off will be to the left of where the turbo warp button is. if turbo warp setting not turned on than the on/off toggle will be put there. everyone can see the control.

**Comment by DNin01** [comment]
It looks like you opened this issue in the wrong repository. I've just moved it to the [ScratchAddons repository](https://github.com/ScratchAddons/ScratchAddons) for you. Thanks for the feedback!

Related to your feedback is #5148, suggesting that the user be able to configure on-screen buttons that act as keys.

**Comment by DNin01** [comment]
I'm not sure if we would the project creators to be responsible for adding a virtual control stick, as users who don't have this addon available may not be able to enjoy those games to their fullest potential. We could, however, make it user-customizable, and perhaps optionally customizable by the creators, like with the Gamepad support addon's "Store the mappings set above inside the project" checkbox.

**Comment by BroJac5246** [comment]
> like with the Gamepad support addon's \"Store the mappings set above inside the project\" checkbox.

This is how it should be, with presets for WASD and arrow keys.

**Comment by mybearworld** [comment]
This might  cause some project creators not to implement their own mobile controls into their project, instead replying on this addon,which of course not everyone will have.

**Comment by DNin01** [comment]
> This might  cause some project creators not to implement their own mobile controls into their project, instead replying on this addon,which of course not everyone will have.

That's a really good point. And since this addon would make it so easy, I'm not sure if simply encouraging creators to program their own would help as much as we want.

**Comment by jack-oid** [comment]
> This might cause some project creators not to implement their own mobile controls into their project, instead replying on this addon,which of course not everyone will have.

true and good pint.

**Comment by BroJac5246** [comment]
> This might  cause some project creators not to implement their own mobile controls into their project, instead replying on this addon,which of course not everyone will have.

Maybe... but the people adding support are considerate people so what's to say they won't continue to be considerate?

Also, only a small subset of Scratchers use the extension, so only a few projects would lose support... but for our users, many would gain support.

**Comment by Secret-chest** [comment]
Yes. This can be said for many addons, it is not a good reason to reject them, our mission is to make the Scratch experience better!

**Comment by DNin01** [comment]
> > This might  cause some project creators not to implement their own mobile controls into their project, instead replying on this addon,which of course not everyone will have.
> 
> Maybe... but the people adding support are considerate people so what's to say they won't continue to be considerate?
> 
> Also, only a small subset of Scratchers use the extension, so only a few projects would lose support... but for our users, many would gain support.

There are all those projects that don't support mobile controls anyway, and this addon makes complete sense there. Maybe our addon would remove a few people's motivation to add mobile controls, but we could just add a note to say, don't expect that everyone is using the addon.

**Comment by jack-oid** [comment]
> > > This might  cause some project creators not to implement their own mobile controls into their project, instead replying on this addon,which of course not everyone will have.
> > 
> > 
> > Maybe... but the people adding support are considerate people so what's to say they won't continue to be considerate?
> > Also, only a small subset of Scratchers use the extension, so only a few projects would lose support... but for our users, many would gain support.
> 
> There are all those projects that don't support mobile controls anyway, and this addon makes complete sense there. Maybe our addon would remove a few people's motivation to add mobile controls, but we could just add a note to say, don't expect that everyone is using the addon.

smart

**Comment by iqnite** [comment]
I wish we already had this addon, but there's a big point we're forgetting here: Scratch Addons is not available on mobile devices, except with some browsers that support CWS on phones like Kiwi browser. Additionally, we'd need to test Scratch Addons on mobile devices.

**Comment by mybearworld** [comment]
> I wish we already had this addon, but there's a big point we're forgetting here: Scratch Addons is not available on mobile devices, except with some browsers that support CWS on phones like Kiwi browser. Additionally, we'd need to test Scratch Addons on mobile devices.

#6789

**Comment by OneShot-Niko** [comment]
> I wish we already had this addon, but there's a big point we're forgetting here: Scratch Addons is not available on mobile devices, except with some browsers that support CWS on phones like Kiwi browser. Additionally, we'd need to test Scratch Addons on mobile devices.

(ok i know im late because mybearworld mentioned it but)

Firefox has already launched support onto extensions (No longer requires Firefox Nightly for Android even), however ScratchAddons currently has zero support for it besides on project pages, but even then thats mainly due to the mobile layout of project pages.

**Comment by BroJac5246** [comment]
> I wish we already had this addon, but there's a big point we're forgetting here: Scratch Addons is not available on mobile devices, except with some browsers that support CWS on phones like Kiwi browser. Additionally, we'd need to test Scratch Addons on mobile devices.

We're already available on Firefox, and Edge is about to get support (https://github.com/ScratchAddons/ScratchAddons/issues/6789#issuecomment-1920121394) — and it seems like that one won't even be opt-in.

**Comment by WorldLanguages** [comment]
Related website: https://scratchgo.app/
